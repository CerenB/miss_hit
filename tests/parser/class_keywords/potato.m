% (c) Copyright 2019 Florian Schanda

classdef potato
  properties
    x
    y
  end

  methods
    function r = foo(self)
      properties = x;
      r = properties;
    end
  end
end
