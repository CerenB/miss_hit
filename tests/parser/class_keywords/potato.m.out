----------------------------------------------------------------------
--  Parse tree for potato.m
Root: Class_File
  Classdef: Class_Definition
    Name: Identifier <potato>
    Properties: Properties Special_Block
      Items: Entity_Constraints
        Name: Identifier <x>
      Items: Entity_Constraints
        Name: Identifier <y>
    Methods: Methods Special_Block
      Items: Function_Definition
        Signature: Function_Signature
          Name: Identifier <foo>
          Inputs: Identifier <self>
          Outputs: Identifier <r>
        Body: Sequence_Of_Statements
          Statements: Simple_Assignment_Statement
            LHS: Identifier <properties>
            RHS: Identifier <x>
          Statements: Simple_Assignment_Statement
            LHS: Identifier <r>
            RHS: Identifier <properties>
      Items: Function_Definition
        Signature: Function_Signature
          Name: Identifier <properties>
          Inputs: Identifier <self>
          Outputs: Identifier <p>
        Body: Sequence_Of_Statements
          Statements: Simple_Assignment_Statement
            LHS: Identifier <p>
            RHS: Number_Literal <123>
----------------------------------------------------------------------
In potato.m, line 7
|   end
|   ^^^ info: this token is not linked to the ast
In potato.m, line 11
|     function r = foo(self)
|                ^ info: this token is not linked to the ast
In potato.m, line 14
|     end
|     ^^^ info: this token is not linked to the ast
In potato.m, line 17
|     function p = properties(self)
|                ^ info: this token is not linked to the ast
In potato.m, line 19
|     end
|     ^^^ info: this token is not linked to the ast
In potato.m, line 20
|   end
|   ^^^ info: this token is not linked to the ast
In potato.m, line 21
| end
| ^^^ info: this token is not linked to the ast
MISS_HIT Summary: 1 file(s) analysed, everything seemes fine
